File: UserList/app/src/main/java/model/MainActivity.java
package model;

import android.os.Bundle;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.userlist.R;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<UserModel> users = generateUsers();

        UserAdapter adapter = new UserAdapter(this, users);
        ListView listView = findViewById(R.id.user_list);
        listView.setAdapter(adapter);
    }

    private List<UserModel> generateUsers() {
        int[] avatars = {R.drawable.avatar1, R.drawable.avatar2, R.drawable.avatar3, R.drawable.avatar4, R.drawable.avatar5};
        String[] firstNames = {"Alice", "Bob", "Charlie", "Diana", "Eve", "Frank", "Grace", "Harry", "Ivy", "Jack"};
        String[] lastNames = {"Johnson", "Smith", "Brown", "Taylor", "Lee", "White", "Harris", "Martin", "Garcia", "Wilson"};
        String[][] countriesAndCities = {
                {"USA", "New York", "Los Angeles", "Chicago", "Houston", "Miami"},
                {"UK", "London", "Manchester", "Birmingham", "Leeds", "Bristol"},
                {"Canada", "Toronto", "Vancouver", "Montreal", "Calgary", "Ottawa"}
        };

        Random random = new Random();
        List<UserModel> users = new ArrayList<>();

        for (int i = 0; i < 100; i++) {
            int avatar = avatars[random.nextInt(avatars.length)];
            String firstName = firstNames[random.nextInt(firstNames.length)];
            String lastName = lastNames[random.nextInt(lastNames.length)];
            long age = 14 + random.nextInt(86); // Random age between 14 and 99
            int countryIndex = random.nextInt(countriesAndCities.length);
            String country = countriesAndCities[countryIndex][0];
            String city = countriesAndCities[countryIndex][random.nextInt(5) + 1];

            users.add(new UserModel(avatar, firstName, lastName, age, country, city));
        }

        return users;
    }
}

File: UserList/app/src/main/java/com/example/userlist/MainActivity.kt
package com.example.userlist

import model.UserModel
import model.UserAdapter
import android.os.Bundle
import android.widget.ListView
import androidx.activity.ComponentActivity

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val users = generateUsers()

        val adapter = UserAdapter(this, users)
        val listView: ListView = findViewById(R.id.user_list)
        listView.adapter = adapter
    }

    private fun generateUsers(): List<UserModel> {
        val avatars = listOf(
            R.drawable.avatar1, R.drawable.avatar2, R.drawable.avatar3,
            R.drawable.avatar4, R.drawable.avatar5
        )
        val firstNames = listOf("Alice", "Bob", "Charlie", "Diana", "Eve", "Frank", "Grace", "Harry", "Ivy", "Jack")
        val lastNames = listOf("Johnson", "Smith", "Brown", "Taylor", "Lee", "White", "Harris", "Martin", "Garcia", "Wilson")
        val countriesAndCities = listOf(
            Pair("USA", listOf("New York", "Los Angeles", "Chicago", "Houston", "Miami")),
            Pair("UK", listOf("London", "Manchester", "Birmingham", "Leeds", "Bristol")),
            Pair("Canada", listOf("Toronto", "Vancouver", "Montreal", "Calgary", "Ottawa"))
        )

        val random = java.util.Random()
        val users = mutableListOf<UserModel>()

        repeat(100) {
            val avatar = avatars[random.nextInt(avatars.size)]
            val firstName = firstNames[random.nextInt(firstNames.size)]
            val lastName = lastNames[random.nextInt(lastNames.size)]
            val age = (14..99).random().toLong()
            val (country, cities) = countriesAndCities[random.nextInt(countriesAndCities.size)]
            val city = cities[random.nextInt(cities.size)]

            users.add(UserModel(avatar, firstName, lastName, age, country, city))
        }

        return users
    }
}